AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -318,6 +325,7 @@ - (void)_populateFromEntireDataSourceWithMutableNodes:(NSMutableArray *)nodes mu
 
 - (void)beginUpdates
 {
+  [_editingTransactionQueue waitUntilAllOperationsAreFinished];
   // Begin queuing up edit calls that happen on the main thread.
   // This will prevent further operations from being scheduled on _editingTransactionQueue.
   // It's fine if there is an in-flight operation on _editingTransactionQueue,
```

Comment:
I believe this is important so that the delegate itself isn't proceeding with edits to its own state (e.g. that which is returned in the asyncDataSource/Delegate protocols) before any prior ones have finished.